contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,435 @@
+       require(s_subscriptions[subId].owner != address(0), "invalid subId");
+       require(minimumRequestConfirmations >= s_config.minimumRequestBlockConfirmations, "minconfs too low");
+       bool validConsumer;
+       for (uint16 i = 0; i < s_subscriptions[subId].consumers.length; i++) {
+           if (s_subscriptions[subId].consumers[i] == msg.sender) {
+               validConsumer = true;
+               break;
+           }
+       }
+       require(validConsumer, "invalid consumer");
+       require(s_serviceAgreements[keyHash] != address(0), "must be a registered key");
+       uint256 nonce = s_nonces[keyHash][msg.sender] + 1;
+       uint256 preSeedAndRequestId = uint256(keccak256(abi.encode(keyHash, msg.sender, nonce)));
+       // Min req confirmations not needed as part of fulfillment, leave out of the commitment
+       s_callbacks[preSeedAndRequestId] = keccak256(abi.encodePacked(preSeedAndRequestId, block.number, subId, callbackGasLimit, numWords, msg.sender));
+       emit RandomWordsRequested(keyHash, preSeedAndRequestId, subId, minimumRequestConfirmations, callbackGasLimit, numWords, msg.sender);
+       s_nonces[keyHash][msg.sender] = nonce;
+       return preSeedAndRequestId;
+    }
+
+    function getCallback(uint256 requestId)
+    external
+    view
+    returns (bytes32){
+        return s_callbacks[requestId];
+    }
+
+    // Offsets into fulfillRandomnessRequest's _proof of various values
+    //
+    // Public key. Skips byte array's length prefix.
+    uint256 public constant PUBLIC_KEY_OFFSET = 0x20;
+    // Seed is 7th word in proof, plus word for length, (6+1)*0x20=0xe0
+    uint256 public constant PRESEED_OFFSET = 0xe0;
+
+    function fulfillRandomWords(
+        bytes memory _proof
+    )
+    external
+    {
+        uint256 startGas = gasleft();
+        (bytes32 keyHash, uint256 requestId,
+        uint256 randomness, FulfillmentParams memory fp) = getRandomnessFromProof(_proof);
+
+        uint256[] memory randomWords = new uint256[](fp.numWords);
+        for (uint256 i = 0; i < fp.numWords; i++) {
+            randomWords[i] = uint256(keccak256(abi.encode(randomness, i)));
+        }
+
+        // Prevent re-entrancy. The user callback cannot call fulfillRandomWords again
+        // with the same proof because this getRandomnessFromProof will revert because the requestId
+        // is gone.
+        delete s_callbacks[requestId];
+        bytes memory resp = abi.encodeWithSelector(FULFILL_RANDOM_WORDS_SELECTOR, requestId, randomWords);
+        require(gasleft() > fp.callbackGasLimit, "not enough gas for consumer");
+        (bool success,) = fp.sender.call(resp);
+        // Avoid unused-local-variable warning. (success is only present to prevent
+        // a warning that the return value of consumerContract.call is unused.)
+        (success);
+
+        emit RandomWordsFulfilled(requestId, randomWords, true);
```

Comment:
```suggestion
        emit RandomWordsFulfilled(requestId, randomWords, success);
```
Wouldn't we want to indicate success? What's the point of having the param if it is always true?

Reply:
yeah should be success, leftover as true from testing